crates/pico_macros/src/source.rs

```diff
@@ -0,0 +1,58 @@
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+pub(crate) fn source(_args: TokenStream, item: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(item as DeriveInput);
+
+    let struct_name = input.clone().ident;
+    let key = struct_name.to_string();
```

Comment:
nit can we rename this something clearer, like `struct_name_static_str`? I looked at this and thought that `struct_name` and `key` were redundant at first